spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/ScheduledTaskEndpoint.java

```diff
@@ -0,0 +1,179 @@
+import org.springframework.scheduling.annotation.SchedulingConfigurer;
+import org.springframework.scheduling.config.CronTask;
+import org.springframework.scheduling.config.IntervalTask;
+import org.springframework.scheduling.config.ScheduledTaskRegistrar;
+import org.springframework.scheduling.config.Task;
+import org.springframework.scheduling.config.TriggerTask;
+import org.springframework.scheduling.support.ScheduledMethodRunnable;
+
+/**
+ * {@link Endpoint} to expose scheduled task information.
+ *
+ * @author Yunkun Huang
+ * @since 2.0.0
+ */
+@ConfigurationProperties(prefix = "endpoints.schedules")
+public class ScheduledTaskEndpoint
+		extends AbstractEndpoint<List<ScheduledTaskEndpoint.ScheduledTaskInformation>>
+		implements SchedulingConfigurer {
+	private List<ScheduledTaskInformation> list = new ArrayList<>();
+
+	public ScheduledTaskEndpoint() {
+		super("schedules");
+	}
+
+	/**
+	 * Create a new {@link ScheduledTaskEndpoint} instance.
+	 */
+	@Override
+	public List<ScheduledTaskInformation> invoke() {
+		return this.list;
+	}
+
+	@Override
+	public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
+		this.list.addAll(build(taskRegistrar.getFixedRateTaskList(), ScheduledType.FIXEDRATE));
+		this.list.addAll(build(taskRegistrar.getCronTaskList(), ScheduledType.CRON));
+		this.list.addAll(build(taskRegistrar.getFixedDelayTaskList(), ScheduledType.FIXEDDELAY));
+		this.list.addAll(build(taskRegistrar.getTriggerTaskList(), ScheduledType.TRIGGER));
+	}
+
+	private List<ScheduledTaskInformation> build(final List<? extends Task> tasks, final ScheduledType type) {
+		return tasks.stream().map(task -> {
+			ScheduledTaskInformation scheduledTaskInformation = new ScheduledTaskInformation();
+			scheduledTaskInformation.setType(type);
+			scheduledTaskInformation.setName(getGenericName(task.getRunnable()));
+			if (task instanceof IntervalTask) {
+				IntervalTask intervalTask = (IntervalTask) task;
+				scheduledTaskInformation.setInitialDelay(intervalTask.getInitialDelay());
+				scheduledTaskInformation.setInterval(intervalTask.getInterval());
+			}
+			if (task instanceof CronTask) {
+				CronTask cronTask = (CronTask) task;
+				scheduledTaskInformation.setExpression(cronTask.getExpression());
+			}
+			if (task instanceof TriggerTask) {
+				TriggerTask triggerTask = (TriggerTask) task;
+				scheduledTaskInformation.setTrigger(triggerTask.getTrigger().toString());
+			}
+			return scheduledTaskInformation;
+		}).collect(Collectors.toList());
+	}
+
+	private String getGenericName(Runnable runnable) {
+		if (runnable instanceof ScheduledMethodRunnable) {
+			return ((ScheduledMethodRunnable) runnable).getMethod().toGenericString();
+		}
+		return runnable.toString();
+	}
+
+	/**
+	 * Information for one scheduled task.
+	 */
+	@JsonInclude(Include.NON_EMPTY)
+	public class ScheduledTaskInformation {
+		private ScheduledType type;
+		private long interval;
+		private long initialDelay;
```

Comment:
If you switch `interval` and `initialDelay` to `java.lang.Long`, then you can avoid unnecessary zeroes in the response for CRON tasks.

Before:
```
[
    {
        "Type": "CRON",
        "Interval": 0,
        "InitialDelay": 0,
        "Name": "public void com.foo.Bar.update()",
        "Expression": "0 */5 * * * *",
        "Trigger": "CronSequenceGenerator: 0 */5 * * * *"
    }
]
```

After:
```
[
    {
        "Type": "CRON",
        "Name": "public void com.foo.Bar.update()",
        "Expression": "0 */5 * * * *",
        "Trigger": "CronSequenceGenerator: 0 */5 * * * *"
    }
]
```